stl/inc/format

```diff
@@ -366,58 +366,363 @@ private:
+        const auto _Store_index = _Index_array[_Arg_index]._Index;
+        const auto _Length      = _Get_format_arg_storage_size<_Context, _Ty>;
+
+        _CSTD memcpy(_Storage + _Index_length + _Store_index, _STD addressof(_Val), _Length);
+        _Index_array[_Arg_index]._Type = _Arg_type;
+        if (_Arg_index + 1 < _Num_args) {
+            // Set the starting index of the next arg, as that is dynamic, must be called with increasing index
+            _Index_array[_Arg_index + 1] = _Format_arg_store_packed_index{_Store_index + _Length};
+        }
+    }
+
+    // See [format.arg]/5
+    template <class _Ty>
+    void _Store(const size_t _Arg_index, const _Ty& _Val) noexcept {
+        _Store_impl<typename basic_format_arg<_Context>::handle>(
+            _Arg_index, _Basic_format_arg_type::_Custom_type, basic_format_arg<_Context>::handle(_Val));
+    }
+
+    // clang-format off
+    template <class _Ty>
+        requires integral<_Ty> || floating_point<_Ty>
+    void _Store(const size_t _Arg_index, _Ty _Val) noexcept {
+        // clang-format on
+        if constexpr (is_same_v<_Ty, bool>) {
+            _Store_impl<bool>(_Arg_index, _Basic_format_arg_type::_Bool_type, _Val);
+        } else if constexpr (is_same_v<_Ty, _CharType>) {
+            _Store_impl<_CharType>(_Arg_index, _Basic_format_arg_type::_Char_type, _Val);
+        } else if constexpr (is_same_v<_Ty, char> && is_same_v<_CharType, wchar_t>) {
+            _Store_impl<_CharType>(_Arg_index, _Basic_format_arg_type::_Char_type, static_cast<wchar_t>(_Val));
+        } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(int)) {
+            _Store_impl<int>(_Arg_index, _Basic_format_arg_type::_Int_type, static_cast<int>(_Val));
+        } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned int)) {
+            _Store_impl<unsigned int>(_Arg_index, _Basic_format_arg_type::_UInt_type, static_cast<unsigned int>(_Val));
+        } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(long long)) {
+            _Store_impl<long long>(_Arg_index, _Basic_format_arg_type::_Long_long_type, static_cast<long long>(_Val));
+        } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned long long)) {
+            _Store_impl<unsigned long long>(
+                _Arg_index, _Basic_format_arg_type::_ULong_long_type, static_cast<unsigned long long>(_Val));
+        } else if constexpr (is_same_v<_Ty, float>) {
+            _Store_impl<float>(_Arg_index, _Basic_format_arg_type::_Float_type, _Val);
+        } else if constexpr (is_same_v<_Ty, double>) {
+            _Store_impl<double>(_Arg_index, _Basic_format_arg_type::_Double_type, _Val);
+        } else if constexpr (is_same_v<_Ty, long double>) {
+            _Store_impl<long double>(_Arg_index, _Basic_format_arg_type::_Long_double_type, _Val);
+        } else {
+            _STL_VERIFY(false, "Invalid type passed to _Format_arg_store::_Store");

```

Comment:
I believe that a `static_assert` with `_Always_false` would be better than a runtime `_STL_VERIFY` here.